# Object类

object类是Java 一个比较特殊的类，它是类层次结构的根，位于继承树的顶层，即Java中所有的类从根本上都继承自Object类。它是Java中位于没有父类的类。如果一个类没有使用extends关键字明确标识继承另外一个类，那么这个类就默认继承自Object类，因此，Object类就是Java类层中的最高层类，是所有类的超类。换句话说，Java中任何一个类都是他的子类。由于所有的类都是由Object类衍生出来的，所以Object类中的方法是适用于所有类，因此下面的两种类的定义形式，从根本上讲是完全一样的

```java
public class Person{
....
}
```

等价于

```java
public class Person extends Object{
...
}
```

Object类中常用方法及操作功能如下表所示

| 返回类型 | 方法名称  | 输入参数     | 抛出异常                     | 操作功能                                                     |
| -------- | --------- | :----------- | ---------------------------- | ------------------------------------------------------------ |
| 构造方法 | Object    |              |                              | 创建Object对象                                               |
| Object   | clone     |              | CloneNotSupportedException   | 创建并返回此对象的一个副本                                   |
| boolean  | equals    | Object       |                              | 指示其他对象是否与此对象“相等”                               |
| void     | finalize  |              | Throwable                    | 当垃圾回收器确定不存在对该对象的更多引用时，由对象的垃圾回收器调用此方法 |
| Class    | getClass  |              |                              | 返回此Object的运行时类                                       |
| int      | hashCode  |              |                              | 返回该对象的哈希码值                                         |
| void     | notify    |              | IllegalMonitorStateException | 唤醒在此对象监视器上等待的单个线程                           |
| void     | nottfyAll |              | IllegalMonitorStateException | 唤醒在此对象监视器上等待的所有线程                           |
| void     | wait      |              | InterruptException           | 在其他线程调用此对象的notify()方法或notifyAll方法前，导致当前线程等待 |
| String   | toString  |              |                              | 返回该对象的字符串表示                                       |
| void     | wait      | long         | InterruptException           | 在其他线程调用此对象的notify()方法或notifyAll()方法。或者炒作制定的时间量前，导致当前线程等待 |
| void     | wait      | long，   int | InterruptException           | 其他线程调用此对象的notify()方法或notifyAll()方法，或者其他某个线程中断当前线程，或者已超过某个实际时间量前，导致当前线程等待 |